### PR TITLE
Add tree query api and deleting tree

### PR DESCRIPTION
**Pull request by BartoszChrostowski**
Closes #95

### CODE REVIEW

File: backend/Tests/TreeOfAKind.UnitTests/Trees/PeopleDomain.cs [hunk 1]
```diff
@@ -67,6 +67,7 @@ public void AddPersonToTree_ValidData_PersonIsAdded()
 
             Assert.Single(Tree.People);
             Assert.Equal(person, Tree.People.FirstOrDefault());
+            Assert.Equal(Surname, person.Surname);
```
Comment (Dragemil) [inline backend/Tests/TreeOfAKind.UnitTests/Trees/PeopleDomain.cs:70]: Wow stop, you were using `Lastname` somewhere already

File: backend/Tests/TreeOfAKind.UnitTests/Validators/AddPersonCommandValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+            Assert.True(Validator.Validate(command).IsValid);
+        }
+
+        [Fact]
+        public void Validate_TooLongName_ValidationFails()
+        {
+            var command = Fixture.Create<AddPersonCommand>();
+
+            SetPropertyWithReflection(command, nameof(command.Name),
```
Comment (Dragemil) [inline backend/Tests/TreeOfAKind.UnitTests/Validators/AddPersonCommandValidatorTests.cs:28]: What's that

Comment (Dragemil) [reply backend/Tests/TreeOfAKind.UnitTests/Validators/AddPersonCommandValidatorTests.cs:28]: Now I know and I don't like it, how about setting property _without_ reflection?

File: backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        public ValidatorTestsBase()
+        {
+            Validator = new TValidator();
+        }
+
+        protected void SetPropertyWithReflection<T, TValue>(T instance, string fieldName, TValue value)
+        {
+            var field = typeof(T).GetField($"<{fieldName}>k__BackingField",
+                BindingFlags.Instance | BindingFlags.NonPublic);
```
Comment (Dragemil) [inline backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs:20]: Oh right, that isn't really how integration tests work

Comment (BartoszChrostowski) [reply backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs:20]: Those are unit tests

Comment (BartoszChrostowski) [reply backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs:20]: I just don't want to create an object each time and set all those fields one by one :<
I'm just testing a validator, and those only checks if the field is not empty and etc

Comment (Dragemil) [reply backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs:20]: Won't the validator not validate if there are some `null` fields that shouldn't be null?

Comment (BartoszChrostowski) [reply backend/Tests/TreeOfAKind.UnitTests/Validators/ValidatorTestsBase.cs:20]: The rest is generated

**Pull request merged**